src/utils.ts

```diff
@@ -23,16 +23,60 @@ export async function asyncSplit<T> (arr: T[], callback: (elem: T) => Promise<bo
   }, [[], []])
 }
 
+/**
+ * Utility function that will split array into two groups based on sync callback.
+ * @param array
+ * @param isValid
+ * @return [success, failure] array where first are positives based on callback and second are negatives
+ */
 export function split<T> (array: T[], isValid: (elem: T) => boolean): [T[], T[]] {
   return array.reduce<[T[], T[]]>(([pass, fail], elem) => {
     return isValid(elem) ? [[...pass, elem], fail] : [pass, [...fail, elem]]
   }, [[], []])
 }
 
+/**
+ * Utility function for decoding Solidity's byte32 array.
+ * @param fileReference
+ */
+export function decodeByteArray (fileReference: string[]): string {
+  return fileReference
+    .map(hexToAscii)
+    .join('')
+    .trim()
+    .replace(/\0/g, '') // Remove null-characters
```

Comment:
I'm not trying to question the decision here but it poked me in the eye as I don't understand the reason for it. Do we really want to delete all the zeroes from the ASCII representation?

Reply:
It is not a zero but a null-characters as the comment says ;-) They are represented as `\0`. Fore removing zeros it would be without the `\`.

Reply:
ah yea... oops